src/plugins/syscalls/linux.cpp

```diff
@@ -159,193 +183,213 @@ static std::vector<uint64_t> linux_build_argbuf(vmi_instance_t vmi,
+                arguments.push_back(info->regs->rdi);
             if ( nargs > 1 )
-                args.push_back(info->regs->rsi);
+                arguments.push_back(info->regs->rsi);
             if ( nargs > 2 )
-                args.push_back(info->regs->rdx);
+                arguments.push_back(info->regs->rdx);
             if ( nargs > 3 )
-                args.push_back(info->regs->rcx);
+                arguments.push_back(info->regs->rcx);
             if ( nargs > 4 )
-                args.push_back(info->regs->r8);
+                arguments.push_back(info->regs->r8);
             if ( nargs > 5 )
-                args.push_back(info->regs->r9);
+                arguments.push_back(info->regs->r9);
         }
     }
 
-    return args;
+    return arguments;
 }
 
-static event_response_t linux_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+void linux_syscalls::print_syscall(drakvuf_t drakvuf, drakvuf_trap_info_t* info, std::vector<uint64_t> arguments)
 {
-    struct wrapper* w = (struct wrapper*)info->trap->data;
-    syscalls* s = w->s;
+    auto params = libhook::GetTrapParams<linux_syscall_data>(info);
 
-    if (!drakvuf_check_return_context(drakvuf, info, w->pid, w->tid, 0))
-        return VMI_EVENT_RESPONSE_NONE;
+    this->fmt_args.clear();
+    if (arguments.size() > 0)
+    {
+        for (size_t i = 0; i < arguments.size(); i++)
+        {
+            auto str = this->parse_argument(drakvuf, info, params->sc->args[i], arguments[i]);
+            if (!str.empty())
+                this->fmt_args.push_back(keyval(params->sc->args[i].name, fmt::Estr(str)));
+            else
+            {
+                uint64_t value = this->transform_value(drakvuf, info, params->sc->args[i], arguments[i]);
+                this->fmt_args.push_back(keyval(params->sc->args[i].name, fmt::Xval(value)));
+            }
+        }
+    }
 
-    const syscall_t* sc = w->num < NUM_SYSCALLS_LINUX ? linuxsc::linux_syscalls[w->num] : NULL;
+    char* tmp = drakvuf_get_process_name(drakvuf, info->proc_data.base_addr, false);
+    std::string thread_name = tmp ?: "";
+    g_free(tmp);
+
+    fmt::print(this->m_output_format, "syscall", drakvuf, info,
+        keyval("ThreadName", fmt::Estr(thread_name)),
+        keyval("Module", fmt::Qstr(std::move(info->trap->breakpoint.module))),
+        keyval("vCPU", fmt::Nval(info->vcpu)),
+        keyval("CR3", fmt::Xval(info->regs->cr3)),
+        keyval("Syscall", fmt::Nval((uint64_t)(params->num))),
+        keyval("NArgs", fmt::Nval(params->sc->num_args)),
+        keyval("Type", fmt::Estr(params->type)),
+        this->fmt_args
+    );
+}
 
-    print_sysret(s, drakvuf, info, w->num, info->trap->breakpoint.module, sc, info->regs->rax);
+event_response_t linux_syscalls::linux_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    auto params = libhook::GetTrapParams<linux_syscall_data>(info);
+    if (!drakvuf_check_return_context(drakvuf, info, params->pid, params->tid, params->rsp))
```

Comment:
There's already a helper for this.
```suggestion
    if (!params->verifyResultCallParams(drakvuf, info))
```